lls/main.c

```diff
@@ -16,11 +16,203 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "gatekeeper_lls.h"
+#include <rte_cycles.h>
+#include <rte_ethdev.h>
+
+#include "cache.h"
+
+/* Length of time (in seconds) to wait between scans of the cache. */
+#define LLS_CACHE_SCAN_INTERVAL		10
+
+static struct lls_config lls_conf;
+
+struct lls_config *
+get_lls_conf(void)
+{
+	return &lls_conf;
+}
+
+static int
+cleanup_lls(void)
+{
+	/* TODO Destroy ARP and ND cache. */
+	destroy_mailbox(&lls_conf.mb);
+	rte_timer_stop(&lls_conf.timer);
+	return 0;
+}
+
+static void
+process_pkts(__attribute__((unused)) struct lls_config *lls_conf,
+	struct gatekeeper_if *iface, uint16_t rx_queue,
+	__attribute__((unused)) uint16_t tx_queue)
+{
+	struct rte_mbuf *bufs[GATEKEEPER_MAX_PKT_BURST];
+	uint16_t num_rx = rte_eth_rx_burst(iface->id, rx_queue, bufs,
+		GATEKEEPER_MAX_PKT_BURST);
+	uint16_t i;
+
+	if (num_rx == 0)
+		return;
+
+	for (i = 0; i < num_rx; i++) {
+		struct ether_hdr *eth_hdr = rte_pktmbuf_mtod(bufs[i],
+			struct ether_hdr *);
+
+		if (!is_broadcast_ether_addr(&eth_hdr->d_addr) &&
+		    !is_same_ether_addr(&eth_hdr->d_addr, &iface->eth_addr))
```

Comment:
In the types of bonding that we're using, yes: http://dpdk.org/doc/guides-16.07/prog_guide/link_bonding_poll_mode_drv_lib.html#configuration